packages/wrangler/src/pages/functions/routes-transformation.test.ts

```diff
@@ -0,0 +1,35 @@
+import { convertRoutePathsToGlobPatterns } from "./routes-transformation";
+
+describe("route-paths-to-glob-patterns", () => {
+  describe("convertRoutePathsToGlobPatterns()", () => {
+    it("should pass through routes with no wildcards", () => {
+      expect(convertRoutePathsToGlobPatterns(["/api/foo"])).toEqual(["/api/foo"]);
+      expect(convertRoutePathsToGlobPatterns(["/api/foo", "/api/bar"])).toEqual(["/api/foo", "/api/bar"]);
+      expect(convertRoutePathsToGlobPatterns(["/api/foo", "/api/bar/foo", "/foo/bar"])).toEqual(["/api/foo", "/api/bar/foo", "/foo/bar"]);
+    });
+
+    it("should escalate a single param route to a wildcard", () => {
+      expect(convertRoutePathsToGlobPatterns(["/api/:foo"])).toEqual(["/api/*"]);
```

Comment:
There's a special case we need to add:
`/` should translate to `/*` - a middleware on the root of the site will generate a route with `/`, which should run on every request.